notebooks/llm-chatbot/llm-chatbot-generate-api.ipynb

```diff
@@ -12,8 +12,8 @@
     "Large Language Models (LLMs) are artificial intelligence systems that can understand and generate human language. They use deep learning algorithms and massive amounts of data to learn the nuances of language and produce coherent and relevant responses.\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #3.    model_dir = convert_and_compress_model(model_id, model_configuration, compression_variant.value, use_preconverted.value)
</pre><p>Looks like optimum is broken:</p>

<pre class="ql-syntax" spellcheck="false">⌛ qwen2-0.5b-instruct conversion to INT4 started. It may takes some time.
</pre><p><strong>Export command:</strong></p><p><code>optimum-cli export openvino --model Qwen/Qwen2-0.5B-Instruct --task text-generation-with-past --weight-format int4 --group-size 128 --ratio 0.8 qwen2/INT4_compressed_weights</code></p>

<pre class="ql-syntax" spellcheck="false">Traceback (most recent call last):
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/bin/optimum-cli", line 8, in &lt;module&gt;
    sys.exit(main())
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/optimum_cli.py", line 198, in main
    register_optimum_cli_subcommand(command_or_command_info, parent_command=parent_command_instance)
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/optimum_cli.py", line 174, in register_optimum_cli_subcommand
    parent_command.register_subcommand(command_info)
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/base.py", line 135, in register_subcommand
    self.registered_subcommands.append(command_info.subcommand_class(self.subparsers, command=command_info))
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/base.py", line 99, in __init__
    self.register_subcommand(subcommand)
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/base.py", line 135, in register_subcommand
    self.registered_subcommands.append(command_info.subcommand_class(self.subparsers, command=command_info))
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/neural_compressor/quantize.py", line 63, in __init__
    super().__init__(
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/base.py", line 87, in __init__
    self.parse_args(self.parser)
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/neural_compressor/quantize.py", line 70, in parse_args
    return parse_args_inc_quantize(parser)
  File "/home/maleksandr/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/venv/lib/python3.10/site-packages/optimum/commands/neural_compressor/quantize.py", line 49, in parse_args_inc_quantize
    f" {str(list(TasksManager._TASKS_TO_AUTOMODELS.keys()))}."
AttributeError: type object 'TasksManager' has no attribute '_TASKS_TO_AUTOMODELS'
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
Cell In[15], line 3
      1 from llm_config import convert_and_compress_model
----&gt; 3 model_dir = convert_and_compress_model(model_id, model_configuration, compression_variant.value, use_preconverted.value)

File ~/test_notebooks/review_refactor_chatbot/openvino_notebooks/notebooks/llm-chatbot/llm_config.py:636, in convert_and_compress_model(model_id, model_config, precision, use_preconverted)
    634 display(Markdown("**Export command:**"))
    635 display(Markdown(f"<code>{optimum_cli_command}</code>"))
--&gt; 636 subprocess.run(optimum_cli_command.split(" "), shell=(platform.system() == "Windows"), check=True)
    637 print(f"✅ {precision} {model_id} model converted and can be found in {model_dir}")
    638 return model_dir

File /usr/lib/python3.10/subprocess.py:526, in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    524     retcode = process.poll()
    525     if check and retcode:
--&gt; 526         raise CalledProcessError(retcode, process.args,
    527                                  output=stdout, stderr=stderr)
    528 return CompletedProcess(process.args, retcode, stdout, stderr)

CalledProcessError: Command '['optimum-cli', 'export', 'openvino', '--model', 'Qwen/Qwen2-0.5B-Instruct', '--task', 'text-generation-with-past', '--weight-format', 'int4', '--group-size', '128', '--ratio', '0.8', 'qwen2/INT4_compressed_weights']' returned non-zero exit status 1.
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
It works for me with last optimum from master

Reply:
@aleksandr-mokrov  which optimum version do you have?

Reply:
<p>@eaidova,  I have already resolved this comment. I took your clone master branch, not original. My mistake.</p>